### DISCUSSION TITLE
Is it possible to use a raidz1 created under Truenas with ZFSBootMenu

### DISCUSSION THREAD

**Discussion by Waldorf3** [zbm-dev/zfsbootmenu#536] [Questions] [answered]
I've kinda painted myself into a corner with a Proxmox + Truenas Scale setup, and realize that all I really want/need is a standard Debian server with root on ZFS.

Using zfsbootmenu this ought to be possible, but I have 7 TB on the dataset that I'd rather prefer not having to restore from backup as that takes almost 48 hours, so the question is, is there any way to reuse the existing pool with zfsbootmenu? I mean, it's a zpool, so it should be possible to just add a new OS dataset and install Debian on it?

**Comment by zdykstra** [comment]
It largely depends on the feature flags enabled on your pool. You have to consider two ZFS implementations to know if it'll work: 

* The version of ZFS in ZFSBootMenu - in the case of the pre-made EFI downloads, that's OpenZFS 2.2.0
* The version of ZFS available in Debian - I don't know exactly what they ship these days / what's readily available

Once you know if those two versions can actually import your pool, you'll can just create a new boot environment and go to town. The easiest way to test is to probably boot the relevant Debian live CD and try to import your pool. If that works, then ZFSBootMenu should also be able to import it.

**Reply by Waldorf3** [reply]
The zpool was created with zfs 2.2.0-rc4.  
I believe Debian uses zfs 2.1.11-1.

Does that help?

**Reply by zdykstra** [reply]
ZFS 2.1.x can't import a ZFS 2.2.0 pool, unless pool compatibility was set. Given thay they're using rc4 code, I doubt they enabled that.

**Reply by Waldorf3** [reply]
```
root@truenas[~]# zpool get compatibility
NAME       PROPERTY       VALUE          SOURCE
boot-pool  compatibility  off            default
poolz      compatibility  off            default
```

I guess it ends here?

**Reply by zdykstra** [reply, marked as answer]
Yup. If the pool was created under OpenZFS 2.2.0-rc4, it's not possible for OpenZFS 2.1.11 to import it. You'll have to either find a backported version of ZFS for Debian, or wait 5 years for them to ship 2.2.0.
